src/targets/anvil.js

```diff
@@ -0,0 +1,110 @@
+        const cameraHeight = scene.cameras.main.height;
+        const width = 220;
+        const height = 180;
+        const x = cameraWidth / 2 - width + 20;
+        const y = cameraHeight / 2 - height + 60;
+
+        // Add invisible button for anvil
+        this.sprite = new Button(scene, x, y, width, height);
+        this.sprite.on("pointerdown", (pointer) => {
+            if (pointer.rightButtonDown() && !pointer.leftButtonDown()) {
+                this.createRightClickMenu(pointer.x, pointer.y, this.actions);
+            } else {
+                this.clickTarget();
+            }
+        });
+    }
+
+    async clickTarget() {
+        const inv = this.scene.dashboard.inventory;
+        const chat = this.scene.scene.get(CONSTANTS.SCENES.CHAT);
+
+        // See if bar is selected
+        const selectedIndex = inv.curSelectedItemIndex;
+        if (selectedIndex < 0) {
+            chat.writeText("Select a bar in your inventory first.");
+            return;
+        }
+
+        const hasHammer = inv.getInventoryIndex("Hammer");
+
+        if (hasHammer < 0) {
+            chat.writeText("You need a hammer to work the metal with.");
+            return;
+        }
+
+        const selectedItem = inv.inventory[selectedIndex];
+
+        switch (selectedItem.name) {
+            case "Bronze Bar":
+                this.smith("BronzeDagger");
+                break;
+            default:
+                chat.writeText("The anvil can only be used with the bar selected");
+                break;
+        }
+    }
+
+    // Take bar and turn it into a smithable item
+    async smith(itemName) {
+        const inv = this.scene.dashboard.inventory;
+        const chat = this.scene.scene.get(CONSTANTS.SCENES.CHAT);
+        const item = await getItemClass(itemName, this.scene.dashboard);
+
+        const indices = item.bars.map((bar) => ({
+            value: inv.getInventoryIndex(bar.name),
+            count: bar.count,
+        }));
+        const allExist = indices.every(
+            (index) =>
+                index.value >= 0 && inv.inventory[index.value].numItems >= index.count
+        );
+
+        // All ingredients are in inventory
+        if (allExist) {
+            const added = inv.addToInventory(item, true);
+
+            // Inventory has space, reduce bar count by the required amount
+            if (added) {
+                indices.forEach((index) => {
+                    const obj = inv.inventory[index.value];
+                    obj.setNumItems(obj.numItems - index.count);
+                });
+
+                // Log click for stats
+                this.scene.stats.updateClickedTargetStat();
+                characterData.addSkillXp("smithing", item.xp);
+                this.scene.enemyKilled(item.questName);
+
+                // Show smith animation
+                this.scene.clickAnimation();
+            }
+        } else {
+            chat.writeText(item.smeltingErrorMessage);
```

Comment:
I think this should be smithingErrorMessage

Reply:
I made the change